src/commands/appSettings/AzureWebJobsStoragePromptStep.ts

```diff
@@ -6,52 +6,63 @@
+    public constructor(private readonly _options?: IConnectionPromptOptions) {
         super();
-        this._suppressSkipForNow = suppressSkipForNow;
     }
 
     public async prompt(context: T): Promise<void> {
-        const selectAccount: MessageItem = { title: localize('selectAzureAccount', 'Select storage account') };
-        const useEmulator: MessageItem = { title: localize('userEmulator', 'Use local emulator') };
-        const skipForNow: MessageItem = { title: localize('skipForNow', 'Skip for now') };
+        const connectStorageButton: MessageItem = { title: localize('connectStorageAccount', 'Connect Storage Account') };
+        const useEmulatorButton: MessageItem = { title: useEmulator };
+        const skipForNowButton: MessageItem = { title: skipForNow };
 
-        const message: string = localize('selectAzureWebJobsStorage', 'In order to debug, you must select a storage account for internal use by the Azure Functions runtime.');
+        const message: string = localize('connectAzureWebJobsStorage', 'In order to proceed, you must connect a storage account for internal use by the Azure Functions runtime.');
 
-        const buttons: MessageItem[] = [selectAccount];
-        if (process.platform === 'win32') {
-            // Only show on Windows until Azurite is officially supported: https://github.com/Azure/azure-functions-core-tools/issues/1247
-            buttons.push(useEmulator);
-        }
-        if (!this._suppressSkipForNow) {
-            buttons.push(skipForNow);
+        const buttons: MessageItem[] = [connectStorageButton, useEmulatorButton];
+
+        if (!this._options?.suppressSkipForNow) {
+            buttons.push(skipForNowButton);
         }
 
         const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, ...buttons);
-        if (result === selectAccount) {
-            context.azureWebJobsStorageType = 'azure';
-        } else if (result === useEmulator) {
-            context.azureWebJobsStorageType = 'emulator';
+        if (result === connectStorageButton) {
+            context.azureWebJobsStorageType = ConnectionType.Azure;
+        } else if (result === useEmulatorButton) {
+            context.azureWebJobsStorageType = ConnectionType.NonAzure;
+        } else {
+            context.azureWebJobsStorageType = ConnectionType.None;
         }
 
-        context.telemetry.properties.azureWebJobsStorageType = context.azureWebJobsStorageType || 'skipForNow';
+        context.telemetry.properties.azureWebJobsStorageType = context.azureWebJobsStorageType;
     }
 
-    public shouldPrompt(context: T): boolean {
-        // Only should prompt if no storage account was selected
-        context.azureWebJobsStorageType = (!!context.storageAccount || !!context.newStorageAccountName) ? 'azure' : undefined;
+    public shouldPrompt(context: T & { eventHubConnectionType?: ConnectionTypeValues, sqlDbConnectionType?: ConnectionTypeValues }): boolean {
```

Comment:
The issue with doing the setting logic in `prompt()` is that even if we don't actually display anything, it messes with the step count.

I agree it's gross to have to set things in `shouldPrompt()`. In that case, would it make more sense to set `context.azureWebJobsStorageType`wherever `this._options?.preSelectedConnectionType` was set?

I think having a `prompt()` step that doesn't actually prompt is less intuitive than setting things in `shouldPrompt()` though if we want to consolidate the logic all to this file.

Reply:
> The issue with doing the setting logic in `prompt()` is that even if we don't actually display anything, it messes with the step count.

Do users actually care if they skip from step 1 to 3? Or if we hide the step counter?

Reply:
Hard to say. I think people would think it's a bug or that they somehow skipped a step and something went wrong though. 

We could hide the step counter, but I think it's nice to know when you're on the final step since there's currently no confirmation page.

Reply:
(I'll close the other comments relating to the `shouldPrompt()` having effects so we can keep the discussion in one place)